File: android-native/app/src/main/java/com/example/kpkn/domain/auge/SessionMuscleFilter.kt
package com.example.kpkn.domain.auge

import com.example.kpkn.data.models.ExerciseMuscleInfo
import com.example.kpkn.data.models.InvolvedMuscle
import com.example.kpkn.data.models.MuscleRole

object SessionMuscleFilter {
    fun relevantMusclesFor(exercise: ExerciseMuscleInfo?): List<InvolvedMuscle> {
        if (exercise == null) return emptyList()
        val hasHighAxialLoad = (exercise.ssc ?: 0.0) >= 1.5
        return exercise.involvedMuscles.filter { muscle ->
            when (muscle.role) {
                MuscleRole.PRIMARY -> true
                MuscleRole.SECONDARY -> true
                MuscleRole.STABILIZER -> hasHighAxialLoad
                MuscleRole.NEUTRALIZER -> false
            }
        }
    }
}

File: ios-native/KPKNFit/KPKNFit/Domain/Auge/SessionMuscleFilter.swift
import Foundation

enum SessionMuscleFilter {
    static func relevantMusclesFor(_ exercise: ExerciseMuscleInfo?) -> [InvolvedMuscle] {
        guard let exercise = exercise else { return [] }
        let hasHighAxialLoad = (exercise.ssc ?? 0.0) >= 1.5
        return exercise.involvedMuscles.filter { muscle in
            switch muscle.role {
            case .PRIMARY: return true
            case .SECONDARY: return true
            case .STABILIZER: return hasHighAxialLoad
            case .NEUTRALIZER: return false
            }
        }
    }
}
